spec/controllers/admin/petition_details_controller_spec.rb

```diff
@@ -0,0 +1,181 @@
+          expect(assigns(:petition)).to eq petition
+        end
+
+        it 'responds successfully and renders the petition_details/show template' do
+          get :show, petition_id: petition.id
+          expect(response).to be_success
+          expect(response).to render_template('petition_details/show')
+        end
+      end
+
+      describe 'for a sponsored petition' do
+        it_behaves_like 'viewing a petition in the correct state'
+      end
+
+      describe 'for a pending petition' do
+        before { petition.update_column(:state, Petition::PENDING_STATE) }
+        it_behaves_like 'viewing a petition in the correct state'
+      end
+
+      describe 'for a validated petition' do
+        before { petition.update_column(:state, Petition::VALIDATED_STATE) }
+        it_behaves_like 'viewing a petition in the correct state'
+      end
+
+      shared_examples_for 'trying to view a petition in the wrong state' do
+        it 'raises a 404 error' do
+          expect {
+            get :show, petition_id: petition.id
+          }.to raise_error ActiveRecord::RecordNotFound
+        end
+      end
+
+      describe 'for an open petition' do
+        before { petition.update_column(:state, Petition::OPEN_STATE) }
+        it_behaves_like 'trying to view a petition in the wrong state'
+      end
+
+      describe 'for a rejected petition' do
+        before { petition.update_column(:state, Petition::REJECTED_STATE) }
+        it_behaves_like 'trying to view a petition in the wrong state'
+      end
+
+      describe 'for a hidden petition' do
+        before { petition.update_column(:state, Petition::HIDDEN_STATE) }
+        it_behaves_like 'trying to view a petition in the wrong state'
+      end
+    end
+
+    describe 'PATCH #update' do
+      let(:petition) { FactoryGirl.create(:sponsored_petition, title: 'Old title', action: 'Old action', description: 'Old description') }
+
+      def do_update
+        patch :update,
+              petition_id: petition.id,
+              petition: petition_attributes
+      end
+
+      describe 'with valid params' do
+        let(:petition_attributes) do
+          {
+              title: 'New title',
+              action: 'New action',
+              description: 'New description'
+          }
+        end
+
+        shared_examples_for 'updating a petition in the correct state' do
+          it 'redirects to the edit petition page' do
+            do_update
+            petition.reload
+            expect(response).to redirect_to"https://petition.parliament.uk/admin/petitions/#{petition.id}/edit"
+          end
+
+          it 'updates the petition' do
+            do_update
+            petition.reload
+            expect(petition).to be_present
+            expect(petition.title).to eq('New title')
+            expect(petition.action).to eq('New action')
+            expect(petition.description).to eq('New description')
+          end
+        end
```

Comment:
Ha yes! Didn't think about it and that'll be definitely a better test coverage, will add!